### PR TITLE
Replace main Makefiles with Poetry Runner (poethepoet)

### PR DESCRIPTION
**Pull request by danyeaw**
Replaces the Makefiles with a Poetry runner called poethepoet in order to improve compatibility for building in Windows.

### PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I have read, and I am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/master/CONTRIBUTING.md)
- [X] I have read, and I understand the [Code of Conduct](https://github.com/gaphor/gaphor/blob/master/CODE_OF_CONDUCT.md)

### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [ ] Feature
- [X] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Issue Number: N/A

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -59,10 +59,58 @@ babel = "^2.9.1"
...
+poethepoet = "^0.10.0"
 
 [tool.poetry.scripts]
-gaphor = 'gaphor.ui:main'
-gaphorconvert = 'gaphor.plugins.diagramexport.gaphorconvert:main'
+gaphor = "gaphor.ui:main"
+gaphorconvert = "gaphor.plugins.diagramexport.gaphorconvert:main"
+
+
+[tool.poe.tasks]
```
Comment (amolenaar) [inline pyproject.toml:71]: This `poe` tasks thing is really nice.

File: pyproject.toml [hunk 2]
```diff
@@ -59,10 +59,58 @@ babel = "^2.9.1"
...
+    { "cmd" = "mypy gaphor/SysML" },
+]
+raaml = [
+    { "cmd" = "python gaphor/codegen/codegen.py --sysml_profile models/RAAML.gaphor gaphor/RAAML/raaml.py models/RAAML.override" },
+    { "cmd" = "mypy gaphor/RAAML" },
+    { "cmd" = "isort gaphor/RAAML" },
+    { "cmd" = "black gaphor/RAAML/raaml.py" },
+]
+c4model = [
+    { "cmd" = "python gaphor/codegen/codegen.py --uml_profile models/C4Model.gaphor gaphor/C4Model/raaml.py models/C4Model.override" },
```
Comment (amolenaar) [inline pyproject.toml:101]: typo. `raaml.py` -> `c4model.py`.

(also on line 104)

Comment (danyeaw) [reply pyproject.toml]: Fixed, thanks!

**Comment by danyeaw** [comment]
@amolenaar This one took some time to resolve relative paths with PyInstaller, but should be all set now. Next up, I can start to work on changing the CI to use gvsbuild

**Comment by danyeaw** [comment]
Combining with #912

**Pull request closed without merging**